src/legilogic_ethereum/ethereum_json_rpc.ml

```diff
@@ -95,6 +95,16 @@ module TransactionCondition = struct
            end) : (PersistableS with type t := t))
 end
 
+let some_value_or_null : TokenAmount.t -> TokenAmount.t option =
```

Comment:
This function smells really bad. Can we do without it, by passing options as arguments where needed?

The only place this function would be appropriate would be while reading information from the outside world that adopts this dubious convention.

Reply:
Ok. I agree it was smelling bad and so I restructured the code.
It was not a small work but now the value can be put as optional
in which case it is actually zero.